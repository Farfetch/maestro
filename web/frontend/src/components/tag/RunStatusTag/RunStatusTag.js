import {
  CheckCircleOutlined,
  CloseCircleOutlined,
  StopOutlined,
  SyncOutlined
} from "@ant-design/icons";
import { Tag } from "antd";
import PropTypes from "prop-types";
import React from "react";

import { runStatus as runStatusModel } from "../../../lib/api/models";

/**
 *
 * @param {string} runStatus
 * @returns React.Node
 * https://ant.design/components/tag/
 */
const getTagByStatus = (runStatus) => {
  switch (runStatus) {
    case runStatusModel.RUNNING:
      return (
        <Tag key="processing" icon={<SyncOutlined spin />} color="processing">
          {runStatus}
        </Tag>
      );
    case runStatusModel.ERROR:
      return (
        <Tag key="error" icon={<CloseCircleOutlined />} color="error">
          {runStatus}
        </Tag>
      );
    case runStatusModel.FINISHED:
      return (
        <Tag key="success" icon={<CheckCircleOutlined />} color="success">
          {runStatus}
        </Tag>
      );
    case runStatusModel.STOPPED:
      return (
        <Tag key="success" icon={<StopOutlined />} color="orange">
          {runStatus}
        </Tag>
      );
    default:
      return <Tag key="default">{runStatus}</Tag>;
  }
};

const RunStatusTag = ({ runStatus }) => getTagByStatus(runStatus);

RunStatusTag.propTypes = {
  runStatus: PropTypes.oneOf(Object.values(runStatusModel)).isRequired
};

export default RunStatusTag;
