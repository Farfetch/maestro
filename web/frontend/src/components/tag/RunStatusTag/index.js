import RunStatusTag from "./RunStatusTag";

export default RunStatusTag;
