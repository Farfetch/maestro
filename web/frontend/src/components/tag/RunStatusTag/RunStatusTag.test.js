import { render } from "@testing-library/react";
import React from "react";

import RunStatusTag from "./RunStatusTag";

describe("components/RunStatusTag", () => {
  ["CREATING", "PENDING", "RUNNING", "STOPPED", "FINISHED", "ERROR"].map(
    (runStatus) =>
      test(`should render ${runStatus} badge`, async () => {
        const rendered = render(<RunStatusTag runStatus={runStatus} />);

        const badgeComponent = rendered.container;
        expect(badgeComponent.outerHTML).toMatchSnapshot();
      })
  );
});
