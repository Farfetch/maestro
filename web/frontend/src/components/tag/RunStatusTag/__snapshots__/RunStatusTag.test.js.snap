// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunStatusTag should render CREATING badge 1`] = `"<div><span class=\\"ant-tag\\">CREATING</span></div>"`;

exports[`components/RunStatusTag should render ERROR badge 1`] = `"<div><span class=\\"ant-tag\\">ERROR</span></div>"`;

exports[`components/RunStatusTag should render FINISHED badge 1`] = `"<div><span class=\\"ant-tag ant-tag-success\\"><span role=\\"img\\" aria-label=\\"check-circle\\" class=\\"anticon anticon-check-circle\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"check-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z\\"></path><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z\\"></path></svg></span><span>FINISHED</span></span></div>"`;

exports[`components/RunStatusTag should render PENDING badge 1`] = `"<div><span class=\\"ant-tag\\">PENDING</span></div>"`;

exports[`components/RunStatusTag should render RUNNING badge 1`] = `"<div><span class=\\"ant-tag ant-tag-processing\\"><span role=\\"img\\" aria-label=\\"sync\\" class=\\"anticon anticon-sync anticon-spin\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"sync\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z\\"></path></svg></span><span>RUNNING</span></span></div>"`;

exports[`components/RunStatusTag should render STOPPED badge 1`] = `"<div><span class=\\"ant-tag ant-tag-error\\"><span role=\\"img\\" aria-label=\\"close-circle\\" class=\\"anticon anticon-close-circle\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z\\"></path><path d=\\"M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z\\"></path></svg></span><span>STOPPED</span></span></div>"`;
